<!DOCTYPE html>
    <html>
        <head>
            <title>Async and await</title>
            <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-QWTKZyjpPEjISv5WaRU9OFeRpok6YctnYmDr5pNlyT2bRjXh0JMhjY6hW+ALEwIH" crossorigin="anonymous">
            <style type="text/css">

           
        
            </style>
        </head>
        <body>

            <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.3/dist/js/bootstrap.bundle.min.js" integrity="sha384-YvpcrYf0tY3lHB60NNkmXc5s9fDVZLESaAA55NDzOxhy9GkcIdslK1eN7N6jIeHz" crossorigin="anonymous"></script>
            <script src="https://code.jquery.com/jquery-3.7.1.min.js"></script>
            <script src="https://unpkg.com/axios/dist/axios.min.js"></script>
            <script type="text/javascript">

                function myfunone(){
                    return "Hello sir. i am myfunone";
                }

                console.log(myfunone());

                // => async 
                async function myfuntwo() {
                    return "hello sir, i am myfuntwo";
                }

                console.log(myfuntwo());
                console.log("Good Bye Sir");

                myfuntwo()
                    .then((response)=>console.log(response)); 

                async function myfunfour(){

                    const setpromise = new Promise((resolve,reject)=> {
                        setTimeout(()=>resolve("Hello Moon"),3000);
                    });

                    const error = true; //false

                    if(!error){
                        const response = await setpromise;
                    }else{
                        await Promise.reject(new Error("Something went wrong !!!"));
                    }
                }

                myfunfour()
                    .then((response)=>console.log(response))
                    .catch((err)=>console.log(err));

            </script>
        </body>
    </html>

